internal/connectors/engine/plugins/impl.go

```diff
@@ -206,6 +211,36 @@ func (i *impl) CreateTransfer(ctx context.Context, req models.CreateTransferRequ
 
 	return models.CreateTransferResponse{
 		Payment: payment,
+		PollingTransferID: func() *string {
+			if resp.TransferId == nil {
+				return nil
+			}
+			return &resp.TransferId.Value
+		}(),
+	}, nil
+}
+
+func (i *impl) PollTransferStatus(ctx context.Context, req models.PollTransferStatusRequest) (models.PollTransferStatusResponse, error) {
+	resp, err := i.pluginClient.PollTransferStatus(ctx, &services.PollTransferStatusRequest{
+		TransferId: req.TransferID,
+	})
+	if err != nil {
+		return models.PollTransferStatusResponse{}, err
+	}
+
+	payment, err := grpc.TranslateProtoPayment(resp.Payment)
+	if err != nil {
+		return models.PollTransferStatusResponse{}, err
+	}
+
```

Comment:
Do we not want to handle the nil case here as we do in FetchNextPayments? (at least add error handling if it is unexpected that it's ever nil)

Reply:
I first chose to do it in this function: `grpc.TranslateProtoPayment(resp.Payment)`, but for clarity you're right, I put a check here